### PR TITLE
Bootstrap controller and add CA resource 

### PR DESCRIPTION
**Pull request by ndbhat**
Description of changes:
1. Bootstrap controller
2. CertificateAuthority Resource with custom Status field: CertificateSigningRequest

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: ATTRIBUTION.md
Comment (a-hilaly) [inline ATTRIBUTION.md:1]: Can you copy the content of ATTRIBUTION.md from another controller repository? e.g https://raw.githubusercontent.com/aws-controllers-k8s/s3-controller/main/ATTRIBUTION.md

File: apis/v1alpha1/ack-generate-metadata.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+ack_generate_info:
+<<<<<<< HEAD
+  build_date: "2023-11-10T17:22:09Z"
+=======
+  build_date: "2023-11-10T17:30:39Z"
+>>>>>>> 18b3556 (Add CertificateSigningRequest Status field and e2e tests)
```
Comment (a-hilaly) [inline apis/v1alpha1/ack-generate-metadata.yaml:6]: Can you please resolve this conflict? regenerating the controller would solve it normally.

File: apis/v1alpha1/certificate_authority.go [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+	// All CRS managed by ACK have a common `Status.Conditions` member that
+	// contains a collection of `ackv1alpha1.Condition` objects that describe
+	// the various terminal states of the CR and its backend AWS service API
+	// resource
+	// +kubebuilder:validation:Optional
+	Conditions []*ackv1alpha1.Condition `json:"conditions"`
+	// The base64 PEM-encoded certificate signing request (CSR) for your private
+	// CA certificate.
+	// +kubebuilder:validation:Optional
+	CertificateSigningRequest *string `json:"certificateSigningRequest,omitempty"`
```
Comment (a-hilaly) [inline apis/v1alpha1/certificate_authority.go:114]: Before making a controller release, we need to discuss whether it's safe or not to store a private CA certificate as base64 in the api-server. Maybe we need to extend the controller runtime to start storing these information in [Secret ](https://kubernetes.io/docs/concepts/configuration/secret/)objects instead.

File: templates/hooks/certificate_authority/sdk_create_post_set_output.go.tpl [hunk 3]
```diff
@@ -0,0 +1,8 @@
+    if ko.Status.ACKResourceMetadata != nil && ko.Status.ACKResourceMetadata.ARN != nil {
+        resourceARN := (*string)(ko.Status.ACKResourceMetadata.ARN)
+        csr, err := rm.getCertificateSigningRequest(ctx, *resourceARN)
+        if err != nil {
+            return nil, err
+        }
+        ko.Status.CertificateSigningRequest = csr
+    }
```
Comment (a-hilaly) [inline templates/hooks/certificate_authority/sdk_create_post_set_output.go.tpl:8]: Are CSR available instantly after a CA creation?

File: test/e2e/tests/helper.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+            logging.info(test_ca["Status"])
+            assert test_ca["Status"] == status
+            assert test_ca["Type"] == "ROOT"
+            assert test_ca["CertificateAuthorityConfiguration"]["Subject"]["CommonName"] == "www.example.com"
+            assert test_ca["CertificateAuthorityConfiguration"]["Subject"]["Country"] == "US"
+            assert test_ca["CertificateAuthorityConfiguration"]["Subject"]["Locality"] == "Arlington"
+            assert test_ca["CertificateAuthorityConfiguration"]["Subject"]["Organization"] == "Example Organization"
+            assert test_ca["CertificateAuthorityConfiguration"]["Subject"]["State"] == "Virginia"
+            assert test_ca["CertificateAuthorityConfiguration"]["KeyAlgorithm"] == "RSA_2048"
+            assert test_ca["CertificateAuthorityConfiguration"]["SigningAlgorithm"] == "SHA256WITHRSA"
```
Comment (a-hilaly) [inline test/e2e/tests/helper.py:36]: From L29 to L36: correct me if i'm wrong, but this looks like it's very tied to the test case itself and those values could be different if we change the Spec of the test resource. If that's the case let's move those assertions after L73 `acmpca_validator.assert_certificate_authority(resource_arn, "PENDING_CERTIFICATE")` and keep this function only to assert that the CA was created successfully and and the status is matching.

Comment (a-hilaly) [reply test/e2e/tests/helper.py:36]: Also the `exists=True` parameter is not used in all the methods

**Review by a-hilaly — COMMENTED** [review]
Thanks a lot @ndbhat! This looks good.
I left few comments/questions inline.

File: templates/hooks/certificate_authority/sdk_create_post_set_output.go.tpl (hunk 3 above)
Comment (ndbhat) [reply templates/hooks/certificate_authority/sdk_create_post_set_output.go.tpl:8]: Yes, they are

File: apis/v1alpha1/certificate_authority.go (hunk 2 above)
Comment (ndbhat) [reply apis/v1alpha1/certificate_authority.go:114]: Agreed, my team also had this question

**Review by a-hilaly — APPROVED** [review]
/lgtm

**Pull request merged**